src/install.js

```diff
@@ -1,10 +1,29 @@
 const fs = require('fs-extra')
 const path = require('path')
 const url = require('url')
-const AdmZip = require('adm-zip')
+const StreamZip = require('node-stream-zip')
+
 const axios = require('axios')
 const ProgressBar = require('progress')
+const tunnel = require('tunnel')
+
+let agent
+if (process.env.HTTPS_PROXY) {
+  const httpsProxyObject = process.env.HTTPS_PROXY.split('://')
+  const urlString = httpsProxyObject[1]
+  const urlArray = urlString.split('@')
+  const proxyAuth = urlArray[0]
+  const hostname = urlArray[1].split(':')[0]
+  const port = urlArray[1].split(':')[1]
```

Comment:
I would prefer to initialize a new URL: https://nodejs.org/api/url.html#url_constructor_new_url_input_base

With this, we can more safely and easily access the separate URL parts. E.g:
```javascript
const myURL = new URL('https://example.org/foo#bar');
console.log(myURL.hash);
// Prints #bar

const myURL = new URL('https://example.org:81/foo');
console.log(myURL.host);
// Prints example.org:81
```

Reply:
I've added the new logic. Please review.